File: dev/20190415/k2works/fizzbuzz/src/main/java/org/hiroshima_arc/application/FizzBuzzValueCommand.java
package org.hiroshima_arc.application;

import org.hiroshima_arc.domain.model.IFizzBuzz;
import org.hiroshima_arc.domain.type.FizzBuzzType;

public class FizzBuzzValueCommand implements IFizzBuzzCommand {
    private FizzBuzzType _type;

    public FizzBuzzValueCommand(FizzBuzzType type) {
        _type = type;
    }

    @Override
    public IFizzBuzz execute(int number) {
        return  _type.generate(number);
    }
}

File: dev/20190822/k2works/src/FizzBuzzValueCommand.ts
import { IFizzBuzzCommand } from "./IFizzBuzzCommand";
import { IFizzBuzzType } from "./IFizzBuzzType";

export class FizzBuzzValueCommand implements IFizzBuzzCommand {
  // tslint:disable-next-line: variable-name
  private _type: IFizzBuzzType;

  constructor(type: IFizzBuzzType) {
    this._type = type;
  }
  public execute(input: number): import("./IFizzBuzz").IFizzBuzz {
    return this._type.generate(input);
  }
}
